packages/next/src/server/render.tsx

```diff
@@ -452,15 +452,17 @@ export async function renderToHTMLImpl(
     runtime: globalRuntime,
     isExperimentalCompile,
   } = renderOpts
+
+  if (!renderOpts.Component) {
+    throw new Error('Invariant: Cannot render without a Component')
```

Comment:
Invarients here ensure that we don't have mystery failures due to passing around an `any` which was `undefined`.